### PR TITLE
Base scripts

### PR DESCRIPTION
**Pull request by NikitaZotov**
Scripts is moved from ostis-web-platform and sc-machine

### CODE REVIEW

**Review by MksmOrlov — CHANGES_REQUESTED** [review]
Add readme and scripts documentation

File: submodule-scripts/update_submodule.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+"${SUBMODULE_SCRIPTS_DIR}/remove_submodule.sh" "$@"
+
+case $? in
+  0) 
+    "${SUBMODULE_SCRIPTS_DIR}/install_submodule.sh" "$@"
+    ;;
+  1) 
+    "${SUBMODULE_SCRIPTS_DIR}/install_submodule.sh" "$@"
+    ;;
```
Comment (MksmOrlov) [inline submodule-scripts/update_submodule.sh:14]: what is the difference?

**Comment by FallenChromium** [comment]
I don't see much value in these scripts, since it's not a standard (nor a guarantee) that all our projects use CMake as one and only step for building the project, or that submodules are only installed using `git submodule`. 

Those choices are not always the optimal solution, but the existence of this repo will push people into using this solution rather than what's optimal for the project they're building. Any change to these scripts may break behavior expectations (or customizations and crutches that were made around those expectations, because clearly the scripts in the PR are not extensible enough to cover all the cases). It's applicable vice versa as well - a small change to the project's requirements may make these scripts unusable for the tasks they're supposed to solve.

TL;DR it doesn't make sense to make these quite specific scripts a separate module since the logic of installing submodules and building the project is tightly coupled to the project - and making tightly coupled code modular leads to bugs, inconsistencies and ugly workarounds.

**Review by FallenChromium — CHANGES_REQUESTED** [review]
I don't think this repo should exist in a form that is suggested in this PR
